quickwit-doc-mapper/src/doc_mapper.rs

```diff
@@ -133,4 +137,103 @@ mod tests {
+        let query = doc_mapper.query(schema, &search_request).unwrap();
+        assert_eq!(
+            format!("{:?}", query),
+            r#"TermQuery(Term(type=Json, field=0, path=toto.titi, vtype=Str, "hello"))"#
+        );
+    }
+
+    #[test]
+    fn test_doc_mapper_query_with_json_field_default_search_fields() {
+        let mut doc_mapper_builder = DefaultDocMapperBuilder::default();
+        doc_mapper_builder.field_mappings.push(FieldMappingEntry {
+            name: "json_field".to_string(),
+            mapping_type: FieldMappingType::Json(
+                QuickwitJsonOptions::default(),
+                Cardinality::SingleValue,
+            ),
+        });
+        doc_mapper_builder
+            .default_search_fields
+            .push("json_field".to_string());
+        let doc_mapper = doc_mapper_builder.build().unwrap();
+        let schema = doc_mapper.schema();
+        let search_request = SearchRequest {
+            index_id: "quickwit-index".to_string(),
+            query: "toto.titi:hello".to_string(),
+            search_fields: vec![],
+            start_timestamp: None,
+            end_timestamp: None,
+            max_hits: 10,
+            start_offset: 0,
+            sort_order: None,
+            sort_by_field: None,
+            aggregation_request: None,
+        };
+        let query = doc_mapper.query(schema, &search_request).unwrap();
+        assert_eq!(
+            format!("{:?}", query),
+            r#"TermQuery(Term(type=Json, field=0, path=toto.titi, vtype=Str, "hello"))"#
+        );
+    }
+
+    #[test]
+    fn test_doc_mapper_query_with_json_field_ambiguous_term() {
+        let doc_mapper_builder = DefaultDocMapperBuilder {
+            field_mappings: vec![FieldMappingEntry {
+                name: "json_field".to_string(),
+                mapping_type: FieldMappingType::Json(
+                    QuickwitJsonOptions::default(),
+                    Cardinality::SingleValue,
+                ),
+            }],
+            default_search_fields: vec!["json_field".to_string()],
+            ..Default::default()
+        };
+        let doc_mapper = doc_mapper_builder.build().unwrap();
+        let schema = doc_mapper.schema();
+        let search_request = SearchRequest {
+            index_id: "quickwit-index".to_string(),
+            query: "toto:5".to_string(),
+            search_fields: vec![],
+            start_timestamp: None,
+            end_timestamp: None,
+            max_hits: 10,
+            start_offset: 0,
+            sort_order: None,
+            sort_by_field: None,
+            aggregation_request: None,
+        };
+        let query = doc_mapper.query(schema, &search_request).unwrap();
+        assert_eq!(
+            format!("{:?}", query),
+            r#"BooleanQuery { subqueries: [(Should, TermQuery(Term(type=Json, field=0, path=toto, vtype=U64, 5))), (Should, TermQuery(Term(type=Json, field=0, path=toto, vtype=Str, "5")))] }"#
+        );
+    }
 }
```

Comment:
Currently, we can query a text field `body` with the query `body.wrong_subpath:hello`. 

Here is the test that is passing:

```rust
#[test]
    fn test_doc_mapper_accept_sub_field_query_on_non_json_field() {
        let mut doc_mapper_builder = DefaultDocMapperBuilder::default();
        doc_mapper_builder.field_mappings.push(FieldMappingEntry {
            name: "json_field".to_string(),
            mapping_type: FieldMappingType::Json(
                QuickwitJsonOptions::default(),
                Cardinality::SingleValue,
            ),
        });
        doc_mapper_builder.field_mappings.push(FieldMappingEntry {
            name: "body".to_string(),
            mapping_type: FieldMappingType::Text(
                QuickwitTextOptions::default(),
                Cardinality::SingleValue,
            ),
        });
        
        let doc_mapper = doc_mapper_builder.build().unwrap();
        let schema = doc_mapper.schema();
        let search_request = SearchRequest {
            index_id: "quickwit-index".to_string(),
            query: "body.wrong_field:hello".to_string(),
            search_fields: vec![],
            start_timestamp: None,
            end_timestamp: None,
            max_hits: 10,
            start_offset: 0,
            sort_order: None,
            sort_by_field: None,
            aggregation_request: None,
        };
        let query = doc_mapper.query(schema, &search_request).unwrap();
        assert_eq!(
            format!("{:?}", query),
            r#"TermQuery(Term(type=Json, field=1, vtype=Str, "hello"))"#
        );
    }
```